### PR TITLE
enable substituting nginx.conf

### PR DESCRIPTION
**Pull request by TomasTomecek**
Please suggest how should I test this feature.

### CODE REVIEW

**Comment by omron93** [comment]
To run test of this new app, `$VERSION/test/run` file has to be modified.

**Comment by TomasTomecek** [comment]
Should be correct now. Sorry it took so long.

Once you ack the design, I'll squash the commits (to two, since the other commit actually fixes the 1.12 fedora dockerfile).

File: 1.12/test/run [hunk 1]
```diff
@@ -236,10 +236,18 @@ cid_file=$(mktemp -u --suffix=.cid)
 
 # Since we built the candidate image locally, we don't want S2I attempt to pull
 # it from Docker hub
-s2i_args="--pull-policy=never"
+s2i_build_args="--pull-policy=never --assemble-user=root"
```
Comment (omron93) [inline 1.12/test/run]: `root` isn't possible in OpenShift. So I'm not sure if we wants this...

File: 1.12/Dockerfile.fedora [hunk 2]
```diff
@@ -58,10 +60,13 @@ COPY ./root/ /
 RUN sed -i -f /opt/app-root/nginxconf-fed.sed /etc/nginx/nginx.conf && \
     mkdir -p /opt/app-root/etc/nginx.d/ && \
     mkdir -p /opt/app-root/etc/nginx.default.d/ && \
+    mkdir -p ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
     chmod -R a+rwx /opt/app-root/etc && \
     chmod -R a+rwx /var /run && \
+    chmod -R a+rwx ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
```
Comment (omron93) [inline 1.12/Dockerfile.fedora]: Why 'a+rwx'? Permissions are too loosen.

Comment (TomasTomecek) [reply 1.12/Dockerfile.fedora]: I just copied from different line w/o thinking too much about it.

File: 1.12/test/run (hunk 1 above)
Comment (TomasTomecek) [reply 1.12/test/run]: I kind of expected this. So can we put the file in app root and just symlink?

File: 1.12/Dockerfile.fedora (hunk 2 above)
Comment (omron93) [reply 1.12/Dockerfile.fedora]: Container should be able to run with default user and with randum UID+root group. `nginx-start` is owned by root group, ug+rwx should be enought there...

File: 1.12/test/run (hunk 1 above)
Comment (omron93) [reply 1.12/test/run]: > . So can we put the file in app root and just symlink?

That's one possible solution. Or we should be able to rewrite it, am I right?

Comment (TomasTomecek) [reply 1.12/test/run]: Yup, that sounds good as well: just let me know which solution I should do.

Comment (omron93) [reply 1.12/test/run]: It's up to you.

I prefer to rewrite file. But I'm not aware of any hard reason for this, so it's totally up to you.

**Comment by TomasTomecek** [comment]
I fixed those two issues (the first one via `chmod a+rwx ${NGINX_CONF_PATH}`) and squashed commits so this PR is ready to be merged from my PoV.

**Comment by omron93** [comment]
Thanks.
[test][test-openshift]

**Pull request merged**